blivet/threads.py

```diff
@@ -33,7 +34,7 @@
 
 def exclusive(m):
     """ Run a callable while holding the global lock. """
-    @wraps(m)
+    @wraps(m, set(functools.WRAPPER_ASSIGNMENTS) & set(dir(m)))
```

Comment:
https://docs.python.org/3/library/functools.html#functools.update_wrapper says:

> The optional arguments are tuples to specify which attributes of the original function are assigned directly to the matching attributes on the wrapper function and which attributes of the wrapper function are updated with the corresponding attributes from the original function.

So we want *everything* found in ``m`` to also be set on the wrapped function? This definitely looks like a change beyond just making this python-23 compatible.